src/query/builder.js

```diff
@@ -932,6 +932,18 @@ assign(Builder.prototype, {
     return this;
   },
 
+  // Remove limit
+  clearLimit() {
+    delete this._single.limit;
```

Comment:
delete is an expensive operation, are you sure that "set to undefined" is not enough in this case?

Reply:
I was thinking the same, that probably setting to null would be better, but then I thought that its premature optimizing and probably doesnt have any difference in the real world.

Reply:
Considering that in this case it doesn't cost anything, I don't see why settle for second best :). And you are right, probably null is better.

Reply:
> delete is an expensive operation, are you sure that "set to undefined" is not enough in this case?

In starts this._single have not props limit and offset. If set {limit: undefined} this._single.hasOwnProperty("limit") will return true. I do not know whether this will lead to conflicts.

Reply:
I don't think "hasOwnProperty" is relevant for anything other than cloning in Knex case and cloning null is fine, so I think it shouldn't be a problem.